Pod/Classes/PINAnimatedImageManager.m

```diff
@@ -380,34 +380,40 @@ + (void)processAnimatedImage:(NSData *)animatedImageData
               });
             }
             
-            CGImageRef frameImage = CGImageSourceCreateImageAtIndex(imageSource, frameIdx, (CFDictionaryRef)@{(__bridge NSString *)kCGImageSourceShouldCache : (__bridge NSNumber *)kCFBooleanFalse});
-            if (frameImage == nil) {
-              NSError *frameImageError = [NSError errorWithDomain:kPINAnimatedImageErrorDomain code:PINAnimatedImageErrorImageFrameError userInfo:nil];
-              HANDLE_PROCESSING_ERROR(frameImageError);
-              break;
-            }
-            
             Float32 duration = durations[frameIdx];
             fileDuration += duration;
-            NSData *frameData = (__bridge_transfer NSData *)CGDataProviderCopyData(CGImageGetDataProvider(frameImage));
-            NSAssert(frameData.length == width * height * kPINAnimatedImageComponentsPerPixel, @"data should be width * height * 4 bytes");
+            
             dispatch_group_async(diskGroup, diskWriteQueue, ^{
+              if (PROCESSING_ERROR) {
+                return;
+              }
+              
+              CGImageRef frameImage = CGImageSourceCreateImageAtIndex(imageSource, frameIdx, (CFDictionaryRef)@{(__bridge NSString *)kCGImageSourceShouldCache : (__bridge NSNumber *)kCFBooleanFalse});
+              if (frameImage == nil) {
+                NSError *frameImageError = [NSError errorWithDomain:kPINAnimatedImageErrorDomain code:PINAnimatedImageErrorImageFrameError userInfo:nil];
+                HANDLE_PROCESSING_ERROR(frameImageError);
+              }
+              
+              NSData *frameData = (__bridge_transfer NSData *)CGDataProviderCopyData(CGImageGetDataProvider(frameImage));
+              NSAssert(frameData.length == width * height * kPINAnimatedImageComponentsPerPixel, @"data should be width * height * 4 bytes");
               NSError *frameWriteError = [self writeFrameToFile:fileHandle duration:duration frameData:frameData];
               HANDLE_PROCESSING_ERROR(frameWriteError);
```

Comment:
Similar to what @rahul-malik mentioned, should we only execute this if `frameImage` is non-`nil`?